pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -232,6 +232,7 @@ public BrokerResponseNative reduceOnDataTable(BrokerRequest brokerRequest,
     }
 
     // Set execution statistics.
+    ResultTable resultTable = brokerResponseNative.getResultTable();
```

Comment:
Since ResultTable is only available in SQL mode, I suggest  doing the following

@JsonProperty("resultTable")
public void setResultTable(ResultTable resultTable) {
    _resultTable = resultTable;
   _numRowsResultSet = resultTable.getRows().size()
  }

This will ensure we don't run into NPE. Also, this will populate non zero numRowsResultSet only for SQL mode but that is fine as PQL will be removed anyway

Reply:
updated!